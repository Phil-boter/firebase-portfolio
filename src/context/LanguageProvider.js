import React, { createContext, useContext, useState } from "react";

export const LanguageContext = createContext(null);
