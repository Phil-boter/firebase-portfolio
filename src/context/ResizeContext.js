import { useEffect, useState, createContext } from "react";

export const ResizeContext = createContext();

function debounce(fn, ms) {
    let timer;
    return (cleanUp) => {
        clearTimeout(timer);
        timer = setTimeout((_) => {
            timer = null;
            fn.apply(this, arguments);
        }, ms);
    };
}

export function ResizeProvider({ children }) {
    const [dimensions, setDimensions] = useState({
        height: window.innerHeight,
        width: window.innerWidth,
    });

    console.log("dim in provider", dimensions);

    useEffect(() => {
        const debouncedHandleResize = debounce(function handleResize() {
            setDimensions({
                height: window.innerHeight,
                width: window.innerWidth,
            });
        }, 500);

        window.addEventListener("resize", debouncedHandleResize);

        return function cleanUp() {
            window.removeEventListener("resize", debouncedHandleResize);
        };
    }, [dimensions, setDimensions]);

    const value = { dimensions };

    return (
        <ResizeContext.Provider value={value}>
            {children}
        </ResizeContext.Provider>
    );
}
