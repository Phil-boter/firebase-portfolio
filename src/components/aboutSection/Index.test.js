import React from "react";

import { render, fireEvent } from "@testing-library/react";

import { ResizeProvider } from "../../context/ResizeContext.js";
import { LanguageContext } from "../../context/LanguageProvider.js";
import AboutSection from "./index.js";

let language = "en";

describe("AboutSection Component", () => {
    it("should render AboutSection", () => {
        const { container } = render(
            <ResizeProvider dimensions={{ height: 1000, widht: 1300 }}>
                <LanguageContext.Provider value={language}>
                    <AboutSection language={language} />
                </LanguageContext.Provider>
            </ResizeProvider>
        );
        expect(container.querySelector(".about")).toBeInTheDocument();
    });
    it("should render Link, that redirrct to GitHub if Click", () => {
        const clickToSpy = jest.fn();
        global.open = clickToSpy;
        const { container } = render(
            <ResizeProvider dimensions={{ height: 1000, widht: 1300 }}>
                <LanguageContext.Provider value={language}>
                    <AboutSection language={language} />
                </LanguageContext.Provider>
            </ResizeProvider>
        );
        expect(container.querySelector(".link-github")).toBeInTheDocument();
        fireEvent.click(container.querySelector(".link-github"));
        expect(clickToSpy).toHaveBeenCalled();
        expect(window.open).toBeCalledWith("https://github.com/Phil-boter");
    });
});
