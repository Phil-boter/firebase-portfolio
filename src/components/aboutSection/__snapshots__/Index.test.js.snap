// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutSection Component should render AboutSection correctly and match sbnapshot 1`] = `
<div>
  <section
    class="section about"
  >
    <div
      class="section-image"
    >
      <img
        alt="book with title story of my life and laptop"
        class="section-inner-image"
        src="assets/content-pixie-iXRVqQtBa_8-unsplash.jpg"
      />
    </div>
    <article
      class="section-article"
    >
      <div
        class="main-header about-header"
      >
        <h1>
          ABOUT ME
           
        </h1>
      </div>
      <div>
        <p>
          I am a junior full stack developer with a background in the food industry and automotive service. I started my professional career with an apprenticeship as a brewer and maltster. After finishing the apprenticeship as the best apprentice in Lower Bavaria, I started my education at the Technical University of Berlin to obtain a diploma as a master brewer, which I successfully completed in 2008
          <br />
          <br />
          After that I worked for the ARWE Group, until then europe's largest company in automotive service, as team leader and later on as quality manager.
          <br />
          When the group went bankrupt during the Corona crisis, I was now able to consistently realise my long-held wish for a reorientation. To realise my long-cherished wish I had already gained my first experience of coding at CodeCademy.com and was sure that coding would be my future. 
          <br />
          When I was offered the chance to attend a bootcamp at Spiced Academy to become a full stack web developer, I took it.
          <br />
          <br />
          In this intensive 12 week course I dived into the depths of JavaScript. At Spiced Academy I built my first full stack web applications using third party APIs, frontend frameworks like React.js and Vue.js and backend like Express. After graduating from the Spiced Academy, I completed the web development career path at CodeCademy. 
          <br />
          Since I personally want to develop myself further and development for mobile devices is becoming more and more important, I started developing my first apps with react-native and Google Firebase. So feel free to have a look at some of my code on
        </p>
        <span
          class="link-github contact-section-link"
        >
          <p>
            Github.
          </p>
        </span>
      </div>
    </article>
  </section>
</div>
`;
